src/sources/kafka.rs

```diff
@@ -179,6 +183,23 @@ fn kafka_source(
                                 log.insert(offset_key, Value::from(msg.offset()));
                             }
 
+                            if let Some(headers_key) = &headers_key {
+                                let mut headers_map = BTreeMap::new();
+                                if let Some(headers) = msg.headers() {
+                                    // Using index-based for loop because rdkafka's `Headers` trait
+                                    // does not provide Iterator-based API
+                                    for i in 0..headers.count() {
+                                        let header =
+                                            headers.get(i).expect("header index is out of bounds");
+                                        headers_map.insert(
+                                            header.0.to_string(),
+                                            Bytes::from(header.1.to_owned()).into(),
+                                        );
```

Comment:
I think I'd still prefer this to avoid the `.expect()`:
```suggestion
                                        if let Some(header) = headers.get(i) {
                                            headers_map.insert(
                                                header.0.to_string(),
                                                Bytes::from(header.1.to_owned()).into(),
                                            );
                                        }
```